src/main/scala/devices/debug/Debug.scala

```diff
@@ -722,6 +721,263 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
       }
     }
 
+    //--------------------------------------------------------------
+    // System Bus Access
+    //--------------------------------------------------------------
+    // --- SBCS Registers ---
+
+    sb2tlOpt match {                   
+      case Some(sb2tl) => {
+       
+        /* Flags for setting error flag registers, needed because the number of SBDATA and
+         * SBADDRESS registers depends on system bus widths
+         */
+        val anyAddressWrEn = Wire(init = false.B)
+        val anyDataRdEn    = Wire(init = false.B)
+        val anyDataWrEn    = Wire(init = false.B)
+
+        // --- SBCS Status Register ---
+        val SBCSFieldsReg      = Reg(new SBCSFields())
+        val SBCSFieldsRegReset = Wire(init = (new SBCSFields()).fromBits(0.U))
+        val SBCSRdData         = Wire(init = new SBCSFields().fromBits(0.U)) 
+        val SBCSWrDataVal      = Wire(init = 0.U(32.W))
+        val SBCSWrData         = Wire(init = new SBCSFields().fromBits(SBCSWrDataVal))
+        val SBCSRdEn, SBCSWrEn = Wire(init = false.B)
+
+
+        // --- System Bus Registers ---
+        val SBADDRESS0FieldsReg = Reg(new SBADDRESS0Fields())
```

Comment:
If you ditch the "Fields" and just create `SBADDRESS` and `SBDATA` of type UInt(32.W), you can just treat these all as  Sequences of whatever they are, conditioned by their length. That would probably greatly reduce this code, especially the part below where you are doing the same thing for each one basically.